main.py

```diff
@@ -1,12 +1,12 @@
-import sys  # used for setting recursion limit higher so program doesn't quit after a while
+import functools
+import typing
```

Comment:
This module just lets you add type hints for more stuff, which mostly just helps document an interface in a way that lets the IDE check your work